CMakeLists.txt

```diff
@@ -35,44 +36,8 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELWITHDEBINFO ${PROJECT_BINARY_DIR}/bin)
 # Build with c++14 support.
 set(CMAKE_CXX_STANDARD 14)
 
-# Disable default libraries in visual studio
-if (MSVC)
-    # Default to statically-linked runtime.
-    if("${MSVC_RUNTIME}" STREQUAL "")
-        set(MSVC_RUNTIME "static")
-    endif()
-
-    # Set compiler options.
-    set(variables
-        CMAKE_C_FLAGS_DEBUG
-        CMAKE_C_FLAGS_MINSIZEREL
-        CMAKE_C_FLAGS_RELEASE
-        CMAKE_C_FLAGS_RELWITHDEBINFO
-        CMAKE_CXX_FLAGS_DEBUG
-        CMAKE_CXX_FLAGS_MINSIZEREL
-        CMAKE_CXX_FLAGS_RELEASE
-        CMAKE_CXX_FLAGS_RELWITHDEBINFO
-    )
-    if(${MSVC_RUNTIME} STREQUAL "static")
-        message(STATUS
-          "MSVC -> forcing use of statically-linked runtime."
-        )
-        foreach(variable ${variables})
-            if(${variable} MATCHES "/MD")
-                string(REGEX REPLACE "/MD" "/MT" ${variable} "${${variable}}")
-            endif()
-        endforeach()
-    else()
-        message(STATUS
-            "MSVC -> forcing use of dynamically-linked runtime."
-        )
-        foreach(variable ${variables})
-            if(${variable} MATCHES "/MT")
-                string(REGEX REPLACE "/MT" "/MD" ${variable} "${${variable}}")
-            endif()
-        endforeach()
-    endif()
-endif ()
+# Use statically linked runtime
+set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
```

Comment:
Shouldn't the actual library depend on particular configuration? Cause here we use Debug in any case, even if Release was asked.

Reply:
Not sure if you've misread the generator expression or I'm misunderstanding your concern:
`"MultiThreaded$<$<CONFIG:Debug>:Debug>" will expand to `MultiThreaded` if not `Debug` (so any flavor of `Release`), and `MultiThreadedDebug` if `Debug`. 

To go into more detail of why the old implementation needed to be replaced:

The previous implementation was troublesome as if I moved up the CMake version without setting [CMAKE_MSVC_RUNETIME_LIBRARY](https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html), I would get the unpleasant default multithreaded behavior. Symbols between my `BlankBot main.cpp` code would mismatch against the `cpp-sc2` compiled libraries and nothing would work. The regex implementation also was post-processing compilation flags, which was understandable at the time the commit was made as the runtime library abstraction simply wasn't available in CMake at the time. But now there's a definitive means of setting the runtime configuration.

Reply:
Ah, this is something new in cmake syntax I am not aware of. Thank you, this part looks good.